tests/tests/bitasset_tests.cpp

```diff
@@ -0,0 +1,510 @@
+      op.new_options.flags &= ~witness_fed_asset;
+      trx.operations.push_back(op);
+      sign( trx, nathan_private_key );
+      PUSH_TX( db, trx, ~0 );
+      generate_block();
+      trx.clear();
+   }
+
+   BOOST_TEST_MESSAGE("Create JMJBIT based on USDBIT.");
+   asset_id_type bit_jmj_id = create_bitasset("JMJBIT").id;
+   {
+      BOOST_TEST_MESSAGE("Update the JMJBIT asset options");
+      asset_update_operation op;
+      const asset_object& obj = bit_jmj_id(db);
+      op.asset_to_update = bit_jmj_id;
+      op.issuer = obj.issuer;
+      op.new_issuer = nathan_id;
+      op.new_options = obj.options;
+      op.new_options.flags &= witness_fed_asset;
+      trx.operations.push_back(op);
+      sign(trx, nathan_private_key);
+      PUSH_TX( db, trx, ~0);
+      generate_block();
+      trx.clear();
+   }
+   {
+      BOOST_TEST_MESSAGE("Update the JMJBIT bitasset options");
+      asset_update_bitasset_operation ba_op;
+      const asset_object& obj = bit_jmj_id(db);
+      ba_op.asset_to_update = obj.get_id();
+      ba_op.issuer = obj.issuer;
+      ba_op.new_options.short_backing_asset = bit_usd_id;
+      ba_op.new_options.minimum_feeds = 1;
+      trx.operations.push_back(ba_op);
+      sign(trx, nathan_private_key);
+      PUSH_TX(db, trx, ~0);
+      generate_block();
+      trx.clear();
+   }
+
+   BOOST_TEST_MESSAGE("Grab active witnesses");
+   auto& global_props = db.get_global_properties();
+   std::vector<account_id_type> active_witnesses;
+   for(const witness_id_type& wit_id : global_props.active_witnesses)
+      active_witnesses.push_back(wit_id(db).witness_account);
+   BOOST_REQUIRE_EQUAL(active_witnesses.size(), 10);
+
+   {
+      BOOST_TEST_MESSAGE("Adding price feed 1");
+      const asset_object& bit_jmj = bit_jmj_id(db);
+      const asset_object& bit_usd = bit_usd_id(db);
+      asset_publish_feed_operation op;
+      op.publisher = active_witnesses[0];
+      op.asset_id = bit_jmj_id;
+      op.feed.settlement_price = ~price(bit_usd.amount(1),bit_jmj.amount(300));
+      op.feed.core_exchange_rate = ~price(core.amount(1), bit_jmj.amount(300));
+      trx.operations.push_back(std::move(op));
+      PUSH_TX( db, trx, ~0);
+
+      const asset_bitasset_data_object& bitasset = bit_jmj_id(db).bitasset_data(db);
+      BOOST_CHECK_EQUAL(bitasset.current_feed.settlement_price.to_real(), 300.0);
+      BOOST_CHECK(bitasset.current_feed.maintenance_collateral_ratio == GRAPHENE_DEFAULT_MAINTENANCE_COLLATERAL_RATIO);
+   }
+   {
+      BOOST_TEST_MESSAGE("Adding price feed 2");
+      const asset_object& bit_jmj = bit_jmj_id(db);
+      const asset_object& bit_usd = bit_usd_id(db);
+      asset_publish_feed_operation op;
+      op.publisher = active_witnesses[1];
+      op.asset_id = bit_jmj_id;
+      op.feed.settlement_price = ~price(bit_usd.amount(1),bit_jmj.amount(100));
+      op.feed.core_exchange_rate = ~price(core.amount(1), bit_jmj.amount(100));
+      trx.operations.push_back(std::move(op));
```

Comment:
Note: `trx.operations` is not empty **before** this push, since it hasn't got cleared after last push. Similar cases exist in code below, perhaps can lead to issues.

Reply:
Fixed